docs/integrations/mypy.md

```diff
@@ -46,162 +47,159 @@ But [with the plugin enabled](#enabling-the-plugin), it gives the correct error:
   keyword arguments.
-* If `Config.populate_by_name=True`, the generated signature will use the field names,
-  rather than aliases.
-* If `Config.extra='forbid'` and you don't make use of dynamically-determined aliases, the generated signature
-  will not allow unexpected inputs.
-* **Optional:** If the [`init_forbid_extra` **plugin setting**](#configuring-the-plugin) is set to `True`, unexpected inputs to
-  `__init__` will raise errors even if `Config.extra` is not `'forbid'`.
-* **Optional:** If the [`init_typed` **plugin setting**](#configuring-the-plugin) is set to `True`, the generated signature
-  will use the types of the model fields (otherwise they will be annotated as `Any` to allow parsing).
-
-### Generate a typed signature for `Model.model_construct`
-* The [`model_construct`](../concepts/models.md#creating-models-without-validation) method is an alternative to `__init__`
-  when input data is known to be valid and should not be parsed. Because this method performs no runtime validation,
-  static checking is important to detect errors.
-
-### Respect `Config.frozen`
-* If `Config.frozen` is `True`, you'll get a mypy error if you try to change
-  the value of a model field; cf. [faux immutability](../concepts/models.md#faux-immutability).
-
-### Generate a signature for `dataclasses`
-* classes decorated with [`@pydantic.dataclasses.dataclass`](../concepts/dataclasses.md) are type checked the same as standard Python dataclasses
-* The `@pydantic.dataclasses.dataclass` decorator accepts a `config` keyword argument which has the same meaning as [the `Config` sub-class](../concepts/config.md).
+* If the [`populate_by_name`][pydantic.ConfigDict.populate_by_name] model configuration value is set to
+  `True`, the generated signature will use the field names rather than aliases.
+* The [`init_forbid_extra`](#init_forbid_extra) and [`init_typed`](#init_typed) plugin configuration
+  values can further fine-tune the synthesized `__init__` method.
+
+### Generate a typed signature for `model_construct`
+
+* The [`model_construct`][pydantic.BaseModel.model_construct] method is an alternative to model validation when input data is
+  known to be valid and should not be parsed (see the [documentation](../concepts/models.md#creating-models-without-validation)).
+  Because this method performs no runtime validation, static checking is important to detect errors.
+
+### Support for frozen models
+
+* If the [`frozen`][pydantic.ConfigDict.frozen] configuration is set to `True`, you will get
+  an error if you try mutating a model field (see [faux immutability](../concepts/models.md#faux-immutability))
 
 ### Respect the type of the `Field`'s `default` and `default_factory`
+
 * Field with both a `default` and a `default_factory` will result in an error during static checking.
 * The type of the `default` and `default_factory` value must be compatible with the one of the field.
```

Comment:
I'll note that we could/should follow the definition of [`dataclasses.field`](https://github.com/python/typeshed/blob/a2ee32ba5dced792abf0a771737248549dfcf32c/stdlib/dataclasses.pyi#L148-L149) so that this becomes supported without the plugin as well.

Reply:
https://github.com/pydantic/pydantic/pull/10651